spec/controllers/users_controller_spec.rb

```diff
@@ -5,22 +5,25 @@
 
   context "when user is logged in" do
     let(:logged_in_user) do
-      create(:user, id: 111,
-            constituency: build(:ons_constituency),
-            email: "foo@bar.com")
+      create(:ready_to_swap_user1, id: 111, constituency: build(:ons_constituency))
     end
 
     before do
-      sign_in logged_in_user
+      # Stub out authentication
+      allow(request.env["warden"]).to receive(:authenticate!).and_return(logged_in_user)
+      allow(controller).to receive(:current_user).and_return(logged_in_user)
     end
 
     describe "GET #show" do
       it "returns http success" do
-        swap_with_user = create(:user)
-        logged_in_user.swap_with_user_id(swap_with_user.id)
+        ClimateControl.modify SERVER_HOST_PORT: "localhost:3000" do
```

Comment:
This looks like a nicer way of controlling the test environment, see https://thoughtbot.com/blog/testing-and-environment-variables